File: SDKs/amazonaws.com/codebuild/2016-10-06/go/pkg/models/operations/deletereport.go
package operations

import (
	"openapi/pkg/models/shared"
)

type DeleteReportXAmzTargetEnum string

const (
	DeleteReportXAmzTargetEnumCodeBuild20161006DeleteReport DeleteReportXAmzTargetEnum = "CodeBuild_20161006.DeleteReport"
)

type DeleteReportHeaders struct {
	XAmzAlgorithm     *string                    `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                    `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                    `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                    `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                    `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                    `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                    `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        DeleteReportXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type DeleteReportRequest struct {
	Headers DeleteReportHeaders
	Request shared.DeleteReportInput `request:"mediaType=application/json"`
}

type DeleteReportResponse struct {
	ContentType           string
	DeleteReportOutput    map[string]interface{}
	InvalidInputException *interface{}
	StatusCode            int64
}

File: SDKs/amazonaws.com/codebuild/2016-10-06/typescript/src/sdk/models/operations/deletereport.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum DeleteReportXAmzTargetEnum {
    CodeBuild20161006DeleteReport = "CodeBuild_20161006.DeleteReport"
}


export class DeleteReportHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: DeleteReportXAmzTargetEnum;
}


export class DeleteReportRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: DeleteReportHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.DeleteReportInput;
}


export class DeleteReportResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  deleteReportOutput?: Map<string, any>;

  @SpeakeasyMetadata()
  invalidInputException?: any;

  @SpeakeasyMetadata()
  statusCode: number;
}
